src/variables.ts

```diff
@@ -0,0 +1,336 @@
+    inputPath: ObjectPath;
+    responsePath: ObjectPath;
+    depth: number;
+    varDefNode: VariableDefinitionNode;
+    dependencies: Map<string, (...args: any[]) => any>;
+    errorMessage?: string;
+}
+
+function createSubCompilationContext(
+    context: CompilationContext
+): CompilationContext {
+    return { ...context};
+}
+export function compileVariableParsing(
+    schema: GraphQLSchema,
+    varDefNodes: ReadonlyArray<VariableDefinitionNode>
+): (inputs: { [key: string]: any }) => CoercedVariableValues {
+    const errors = [];
+    const coercedValues: { [key: string]: any } = Object.create(null);
+    let body = `
+    return function getVariables (input) {
+    const errors = [];
+    `;
+    const dependencies = new Map();
+    for (const varDefNode of varDefNodes) {
+        const context: CompilationContext = {
+            varDefNode,
+            depth: 0,
+            inputPath: addPath(undefined, "input"),
+            responsePath: addPath(undefined, "coerced"),
+            dependencies
+        };
+        const varName = varDefNode.variable.name.value;
+        const varType = typeFromAST(schema, varDefNode.type as any);
+        if (!varType || !isInputType(varType)) {
+            // Must use input types for variables. This should be caught during
+            // validation, however is checked again here for safety.
+            errors.push(
+                new (GraphQLError as any)(
+                    `Variable "$${varName}" expected value of type ` +
+                    `"${
+                        varType ? varType : print(varDefNode.type)
+                        }" which cannot be used as an input type.`,
+                    computeLocations([varDefNode.type])
+                )
+            );
+            continue;
+        }
+        if (varDefNode.defaultValue) {
+            // If no value was provided to a variable with a default value,
+            // use the default value.
+            coercedValues[varName] = valueFromAST(varDefNode.defaultValue, varType);
+        }
+
+        const hasValueName = hasValue(addPath(context.inputPath, varName));
+        body += `const ${hasValueName} = Object.prototype.hasOwnProperty.call(${
+            getObjectPath(context.inputPath)}, "${varName}");\n`;
+        context.inputPath = addPath(context.inputPath, varName);
+        context.responsePath = addPath(context.responsePath, varName);
+        body += generateInput(context, varType, varName, hasValueName, false);
+    }
+
+    if (errors.length > 0) {
+        throw errors;
+    }
+
+    body += "if (errors.length > 0) {return {errors, coerced: undefined};}";
+    body += "return {errors: undefined, coerced};\n}";
+    return Function
+        .apply(null, ["coerced", "GraphQLError", "inspect"].concat(Array.from(dependencies.keys())).concat(body))
+        .apply(null, [coercedValues, GraphQLError, inspect].concat(Array.from(dependencies.values())));
+}
+
+const MAX_INT = 2147483647;
```

Comment:
Is this number special in GQL? You can get the max safe int and min safe int using Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER.

Reply:
https://graphql.github.io/graphql-spec/June2018/#sec-Int

Reply:
Use `0x7fffffff`